File: HYPERLEDGER/bfn-network-chaincode/bfn-chaincode2/src/helpers/models/country.ts

import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject('Country')
export class Country {

  @JsonProperty('docType', String)
  public docType?: string | undefined;
  @JsonProperty('countryId', String)
  public countryId: string | undefined;
  @JsonProperty('code', String)
  public code: string | undefined;
  @JsonProperty('name', String)
  public name: string | undefined;
  @JsonProperty('vat', Number)
  public vat: number | 0.0;

}

File: HYPERLEDGER/bfn-network-apps/businesslibrary/lib/data/country.dart
class Country {
  String name, code, countryId;
  double vat;

  Country({this.name, this.code, this.vat, this.countryId});
  Country.fromJson(Map data) {
    this.name = data['name'];
    this.code = data['code'];
    if (data['vat'] is int) {
      this.vat = data['vat'] * 1.0;
    } else {
      this.vat = data['vat'];
    }
    this.countryId = data['countryId'];
  }
  Map<String, dynamic> toJson() => <String, dynamic>{
        'name': name,
        'code': code,
        'vat': vat,
        'countryId': countryId,
      };
}
